File: test-module-java/src/main/java/com/hjl/test_module_java/TestLoginAdapter.java
package com.hjl.test_module_java;

import android.content.Context;
import android.util.Log;

import com.hjl.commonlib.base.ICommonCallback;
import com.hjl.commonlib.base.ILoginAdapter;

import org.json.JSONObject;

/**
 * author: long
 * description please add a description here
 * Date: 2021/12/21
 */
public class TestLoginAdapter implements ILoginAdapter {

    public String TAG = "ZRouter";

    private static TestLoginAdapter sInstance;

    public static synchronized TestLoginAdapter getInstance() {
        if (sInstance == null) {
            sInstance = new TestLoginAdapter();
        }
        return sInstance;
    }

    @Override
    public void login(Context context, JSONObject loginInfo, ICommonCallback callback) {
        Log.i("ZRouter", "java login");
        callback.onFinished(1,"java login",null);
    }

    @Override
    public void logout(Context context, ICommonCallback callback) {
        Log.i("ZRouter", "java logout");
        callback.onFinished(1,"java login",null);
    }


}

File: test-module-kotlin/src/main/java/com/hjl/test_module_kotlin/TestLoginAdapter.kt
package com.hjl.test_module_kotlin

import android.content.Context
import android.util.Log
import com.hjl.commonlib.base.ICommonCallback
import com.hjl.commonlib.base.ILoginAdapter
import org.json.JSONObject

/**
 * author: long
 * description please add a description here
 * Date: 2021/12/21
 */
class TestLoginAdapter : ILoginAdapter{
    override fun login(context: Context, loginInfo: JSONObject, callback: ICommonCallback) {
        Log.i("ZRouter", "kt login")
        callback.onFinished(1,"kt login")
    }

    override fun logout(context: Context?, callback: ICommonCallback) {
        Log.i("ZRouter", "kt logout")
        callback.onFinished(1,"kt logout")
    }


    companion object{
        val instance : TestLoginAdapter by lazy(mode = LazyThreadSafetyMode.SYNCHRONIZED) { TestLoginAdapter() }
    }
}